cpp/include/cudf/strings/repeat_strings.hpp

```diff
@@ -76,15 +84,91 @@ std::unique_ptr<string_scalar> repeat_strings(
 
+/**
+ * @brief Repeat each string in the given strings column by the numbers of times given in another
+ * numeric column.
+ *
+ * An output strings column is generated by repeating each of the input string by a number of times
+ * given by the corresponding row in a @p `repeat_times` numeric column. The computational time can
+ * be reduced if sizes of the output strings are known and provided.
+ *
+ * In special cases:
+ *  - Any null row (from either the input strings column or the `repeat_times` column) will always
+ *    result in a null output string.
+ *  - If any value in the `repeat_times` column is not a positive number and its corresponding input
+ *    string is not null, the output string will be an empty string.
+ *
+ * Note that this function cannot handle the cases when the size of the output column exceeds the
+ * maximum value that can be indexed by size_type (offset_type). In such situations, an exception
+ * may be thrown, or the output result is undefined. As such, the caller is responsible for checking
+ * output overflow to prevent runtime exception and data corruption.
+ *
+ * @code{.pseudo}
+ * Example:
+ * strs         = ['aa', null, '', 'bbc-']
+ * repeat_times = [ 1,   2,     3,  4   ]
+ * out          = repeat_strings(strs, repeat_times)
+ * out is ['aa', null, '', 'bbc-bbc-bbc-bbc-']
+ * @endcode
+ *
+ * @throw cudf::logic_error if the input `repeat_times` column has data type other than integer.
+ * @throw cudf::logic_error if the input columns have different sizes.
+ *
+ * @param input The column containing strings to repeat.
+ * @param repeat_times The column containing numbers of times that the corresponding input strings
+ *        are repeated.
+ * @param output_strings_sizes The optional column containing pre-computed sizes of the output
+ *        strings.
+ * @param mr Device memory resource used to allocate the returned strings column.
+ * @return New column containing the repeated strings.
+ */
+std::unique_ptr<column> repeat_strings(
+  strings_column_view const& input,
+  column_view const& repeat_times,
+  std::optional<column_view> output_strings_sizes = std::nullopt,
+  rmm::mr::device_memory_resource* mr             = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Compute sizes of the output strings if each string in the input strings column
+ * is repeated by the numbers of times given in another numeric column.
+ *
+ * During calculating the string output sizes, these sizes are also sum up (stored in an int64_t
+ * number) and returned, which can be used to detect if the input strings column can be safely
+ * repeated without data corruption due to overflow in string indexing.
+ *
+ * @code{.pseudo}
+ * Example:
+ * strs         = ['aa', null, '', 'bbc-']
+ * repeat_times = [ 1,   2,     3,  4   ]
+ * out          = repeat_strings_output_sizes(strs, repeat_times)
+ * out is [2, 0, 0, 16]
+ * @endcode
+ *
```

Comment:
Will the output column contain any nulls? I think it would be easier if it didn't. Either way, it would be good to mention that here.

Reply:
Thanks. The output will not be nullable. I'll add comment for that.